### PR TITLE
XW-4707 | recommended videos - autoplay variation

### PR DESCRIPTION
**Pull request by latata**
## Links

* http://wikia-inc.atlassian.net/browse/XW-4707

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: app/components/recommended-video.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	playlistItem: null,
+	playlistItems: null,
+
+	init() {
+		this._super(...arguments);
+
+		run.later(() => {
+			this.initRecommendedVideo();
+		}, 5000);
```
Comment (rybmat) [inline app/components/recommended-video.js]: did we get this number from designer/product or it is just example one?

File: app/styles/component/_recommended-video-lazy.scss [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	}
+
+	header {
+		align-items: center;
+		display: flex;
+		justify-content: space-between;
+		margin-bottom: 10px;
+
+		.wds-icon-tiny {
+			fill: #ccc;
```
Comment (rybmat) [inline app/styles/component/_recommended-video-lazy.scss]: `$wds-color-light-c4`

File: app/styles/component/_recommended-video-lazy.scss [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		li {
+			display: flex;
+			margin-bottom: 9px;
+
+			&.current {
+				.recommended-video__playlist-thumbnail {
+					padding: 0;
+
+					img {
+						border: 3px solid #00d6d6;
```
Comment (rybmat) [inline app/styles/component/_recommended-video-lazy.scss]: `$wds-fandom-color-aqua`

File: app/styles/component/_recommended-video-lazy.scss [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+		}
+	}
+
+	&__playlist-thumbnail {
+		margin-right: 12px;
+		padding: 2px;
+		position: relative;
+
+		img {
+			border: 1px solid #ccc;
```
Comment (rybmat) [inline app/styles/component/_recommended-video-lazy.scss]: $wds-color-light-c4

File: app/styles/component/_recommended-video-lazy.scss [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+	&__playlist-title {
+		align-items: center;
+		display: flex;
+		font-weight: bold;
+	}
+
+	&__upnext {
+		align-items: center;
+		color: #999;
```
Comment (rybmat) [inline app/styles/component/_recommended-video-lazy.scss]: `$wds-color-light-c5`

File: app/styles/component/_recommended-video-lazy.scss [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+		display: none;
+		font-size: 14px;
+		font-weight: normal;
+		margin: 6px 0 20px;
+		padding: 0 18px;
+
+		&::after {
+			background-color: #ccc;
```
Comment (rybmat) [inline app/styles/component/_recommended-video-lazy.scss]: `$wds-color-light-c4`

**Review by rybmat — CHANGES_REQUESTED** [review]

File: app/components/recommended-video.js (hunk 1 above)
Comment (latata) [reply app/components/recommended-video.js]: it's just an example value.

**Review by rybmat — APPROVED** [review]

File: app/components/recommended-video.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		play(index) {
+			this.get('playerInstance').playlistItem(index);
+		},
+
+		close() {
+			this.setProperties({
+				isClosed: true,
+				noScroll: false
+			});
+			this.isClosed = true;
```
Comment (hakubo) [inline app/components/recommended-video.js]: I think this is not needed

File: app/components/recommended-video.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+	playerCreated(playerInstance) {
+		playerInstance.once('mute', () => {
+			this.setProperties({
+				isExtended: true,
+				noScroll: true
+			});
+		});
+
+		playerInstance.on('playlistItem', ({item}) => {
+			this.set('playlistItem', extend({}, item));
```
Comment (hakubo) [inline app/components/recommended-video.js]: I'd add a comment why you copy an item here

File: app/components/recommended-video.js (hunk 7 above)
Comment (latata) [reply app/components/recommended-video.js]: upsi

File: app/components/recommended-video.js [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+			});
+			this.get('playerInstance').remove();
+		}
+	},
+
+	initRecommendedVideo() {
+		Promise.all([
+			jwPlayerAssets.load(),
+			this.getVideoData()
+		]).then(([, videoData]) => {
```
Comment (xkxd) [inline app/components/recommended-video.js]: order of promises can be changed to get rid of this fancy destructuring

File: app/components/recommended-video.js (hunk 8 above)
Comment (bkoval) [reply app/components/recommended-video.js]: Done

**Pull request merged**